AppInspector/Commands/AnalyzeCommand.cs

```diff
@@ -275,467 +281,253 @@ private void ConfigRules()
+                        WriteOnce.Error($"Overflowed while extracting file entries. Check the input for quines or zip bombs. {e.Message}");
+                    }
+                    done = true;
+                });
 
-            //zip itself may be too huge for timely processing
-            if (new FileInfo(filePath).Length > WARN_ZIP_FILE_SIZE)
-            {
-                if (topLevel)
-                {
-                    WriteOnce.General(MsgHelp.GetString(MsgHelp.ID.ANALYZE_COMPRESSED_FILESIZE_WARN));
-                }
-                else
-                {
-                    WriteOnce.SafeLog("Decompressing large file " + filePath, LogLevel.Warn);
-                }
-            }
-            else
-            {
-                if (topLevel)
-                {
-                    WriteOnce.General(MsgHelp.GetString(MsgHelp.ID.ANALYZE_COMPRESSED_PROCESSING));
-                }
-                else
+                var options = new ProgressBarOptions
                 {
-                    WriteOnce.SafeLog("Decompressing file " + filePath, LogLevel.Warn);
-                }
-            }
-
-            LastUpdated = File.GetLastWriteTime(filePath);
-            _ = _metaDataHelper?.PackageTypes.TryAdd(MsgHelp.GetString(MsgHelp.ID.ANALYZE_COMPRESSED_FILETYPE),0);
-
-            try
-            {
-                var extractor = new Extractor();
-                IEnumerable<FileEntry> files = extractor.ExtractFile(filePath);
+                    ForegroundColor = ConsoleColor.Yellow,
+                    ForegroundColorDone = ConsoleColor.DarkGreen,
+                    BackgroundColor = ConsoleColor.DarkGray,
+                    BackgroundCharacter = '\u2593',
+                    DisableBottomPercentage = true
+                };
 
-                if (_options.SingleThread)
+                using (var pbar = new IndeterminateProgressBar("Enumerating Files.", options))
                 {
-                    foreach (FileEntry file in files)
+                    while (!done)
                     {
-                        try
-                        {
-                            //check uncompressed file passes standard checks
-                            LanguageInfo languageInfo = new LanguageInfo();
-                            if (FileChecksPassed(file, ref languageInfo))
-                            {
-                                var streamByteArray = new byte[file.Content.Length];
-                                file.Content.Read(streamByteArray);
-                                ProcessInMemory(file.FullPath, Encoding.UTF8.GetString(streamByteArray), languageInfo);
-                            }
-                        }
-                        catch (Exception e)
-                        {
-                            WriteOnce.SafeLog($"Failed to parse {file.FullPath}. {e.GetType()}:{e.Message}", LogLevel.Info);
-                        }
+                        Thread.Sleep(10);
```

Comment:
Is there a better way of avoiding this?

Reply:
What do you want to avoid? This isn't really stalling it's just looping to write the number of results to the screen.

Reply:
Ahh, my mistake then! :-)